moveit_ros/moveit_servo/demos/cpp_interface/demo_joint_jog.cpp

```diff
@@ -87,10 +94,19 @@ int main(int argc, char* argv[])
   std::chrono::seconds time_elapsed(0);
   auto start_time = std::chrono::steady_clock::now();
 
+  // create command queue to build trajectory message
+  std::deque<KinematicState> joint_cmd_rolling_window;
+  KinematicState current_state = servo.getCurrentRobotState();
+  current_state.time = demo_node->now();
+  joint_cmd_rolling_window.push_back(current_state);
+
   RCLCPP_INFO_STREAM(demo_node->get_logger(), servo.getStatusMessage());
   while (rclcpp::ok())
   {
-    const KinematicState joint_state = servo.getNextJointState(joint_jog);
+    auto last_commanded_state = joint_cmd_rolling_window.back();
+    robot_state->setJointGroupPositions(joint_model_group, last_commanded_state.positions);
+    KinematicState joint_state = servo.getNextJointState(robot_state, joint_jog);
+    joint_state.time = demo_node->now() + rclcpp::Duration::from_seconds(servo_params.max_expected_latency);
```

Comment:
I simplified the demos. You don't need to add joint states manually. One reason it might seem more complex now is that you have to set the robot state yourself. This makes the interface more functional of course at the cost of more effect on the caller's part.